### PR TITLE
ZAI Test Refactoring

### PR DESCRIPTION
**Pull request by krakjoe**
### Description

This refactors abstract interface tests to use harness for consistent naming and tagging.

Along with that, I've refactored some code into different files - in general the test suite displayed should have some obvious relation to the file where a failing test will be found.

I have not refactored interfaces destined to be replaced by #1452 

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.

### CODE REVIEW

**Comment by bwoebi** [comment]
Feels weird to have __ZAI_SAPI_TEST_CASE_WITHOUT_BAILOUT_BEGIN with leading double underscore, though it's obviously "public API" of the test header?

File: zend_abstract_interface/config/tests/default.cc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+#include "config/config.h"
+#include "zai_sapi/zai_sapi.h"
+#include "zai_sapi/zai_sapi_extension.h"
+#include "ext_zai_config.h"
+}
+
+#include "zai_tests_common.hpp"
+
+TEST_DEFAULT("bool", {
+    REQUEST_BEGIN()
```
Comment (bwoebi) [inline zend_abstract_interface/config/tests/default.cc:13]: this seems repetitive, should be part of the TEST_DEFAULT?

Comment (krakjoe) [reply zend_abstract_interface/config/tests/default.cc:15]: Not without another few macros, so I left it ... but did define TEST_DEFAULT (and others)

File: zend_abstract_interface/config/tests/zai_tests_common.hpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ext_zai_config_ctor(&zai_sapi_extension, PHP_MINIT(zai_config_env)); \
+    REQUIRE(zai_sapi_minit()); \
+    { __VA_ARGS__ }            \
+    zai_sapi_mshutdown();      \
+    zai_sapi_sshutdown();      \
+}
+
+#define TEST_DEFAULT(description, ...) ZAI_SAPI_TEST_CASE_BARE("config/default", description, TEST_BODY(__VA_ARGS__))
+#define TEST_ENV(description, ...)     ZAI_SAPI_TEST_CASE_BARE("config/env", description, TEST_BODY(__VA_ARGS__))
+#define TEST_ID(description, ...)      ZAI_SAPI_TEST_CASE_BARE("config/id", description, TEST_BODY(__VA_ARGS__))
```
[the remark this answers is not in the document]
Comment (krakjoe) [reply zend_abstract_interface/config/tests/zai_tests_common.hpp]: They are now, thanks

**Review by bwoebi — APPROVED** [review]
Cool, looks great to me, more readable tests <3

**Pull request merged**